nomad/eval_broker.go

```diff
@@ -724,6 +754,38 @@ func (b *EvalBroker) ResumeNackTimeout(evalID, token string) error {
 	return nil
 }
 
+func (b *EvalBroker) handleAckNackLocked(eval *structs.Evaluation) {
```

Comment:
There's some potential weirdness here to consider. `Ack` and `Nack` are called with a string ID, but the eval may not exists in the `unack` queue, so I wonder if there's a possibility where `enqueueTime` and `dequeueTime` have an eval ID that is not in `unack`, causing them to never get cleaned up.

At some point I thought of passing `evalID` and `eval` here to handle the clean up but I was being lazy and dropped it 😅